internal/provider/provider.go

```diff
@@ -23,7 +23,7 @@ const (
 	defaultAssumeRoleDuration = 1 * time.Hour
 )
 
-func New() tfsdk.Provider {
+func New() tfsdk.ProviderWithProviderMeta {
```

Comment:
We can still return `tfsdk.Provider` here. As long as `GetMetaSchema` is defined on `AwsCloudControlApiProvider`, it will still fulfill the `tfsdk.ProviderWithProviderMeta` interface